neofs-blocks.mediawiki

```diff
@@ -0,0 +1,58 @@
+<pre>
+  NEP: TBD
+  Title: NeoFS block storage format
+  Author: Ekaterina Pavlova <ekt@nspcc.ru>
+  Type: Informational
+  Status: Draft
+  Created: 2025-04-07
+</pre>
+
+==Abstract==
+
+This proposal outlines the methodology and specifications for storing Neo blockchain blocks within
+the NeoFS container.
+
+==Motivation==
+
+Neo node synchronization via P2P requires all block headers (~700 bytes each, over 1 GB per year with
+15-second blocks) to remain available, limiting scalability as chains grow. Storing blocks in NeoFS
+provides a distributed alternative, reducing local storage, enabling on-demand fetches, and speeding
+up sync with snapshots or checkpoints. This proposal defines a NeoFS storage schema, building on
+experiments (e.g., nspcc-dev/neo-go#3496) and future enhancements (#1526).
+
+==Specification==
+
+===Block Storage Schema===
+A single NeoFS container is used to store blocks. Each container
+has network magic attribute (`Magic:56753`). Each block is stored in a binary
+form as a separate object with a unique OID and a set of attributes:
+ - block object identifier with block index value (`Block:1`)
+ - primary node index (`Primary:0`)
```

Comment:
We have it now, but I'm still thinking if its useful. I can imagine searches for block hashes/timestamps, but primary index is not so obvious (especially without a verification script hash).